Added Release Notes v5.5 (#61)
hazelcast/hazelcast-platform-operator-docs: docs/modules/ROOT/pages/release-notes.adoc

= 5.5 Release Notes

> In this release, we introduced the Custom Resources for xref:topic-configuration.adoc[Topic], xref:multimap-configuration.adoc[Multimap] and xref:replicatedmap-configuration.adoc[Replicated Map] data structure support. We also introduced EntryListener support for Map CR.

Reviewer: ```suggestion
In this release, we introduced custom resources (CRs) for xref:topic-configuration.adoc[Topic], xref:multimap-configuration.adoc[Multimap], and xref:replicatedmap-configuration.adoc[Replicated Map] data structures. We also added EntryListener support for the Map CR.
```

== New features

> * Add support for MultiMap CR (https://github.com/hazelcast/hazelcast-platform-operator/pull/367[#367])

Reviewer: ```suggestion
* Added support for the MultiMap CR (https://github.com/hazelcast/hazelcast-platform-operator/pull/367[#367])
```

> * Add support for Topic CR (https://github.com/hazelcast/hazelcast-platform-operator/pull/371[#371])

Reviewer: ```suggestion
* Added support for Topic CR (https://github.com/hazelcast/hazelcast-platform-operator/pull/371[#371])
```

> * Added support for Topic CR (https://github.com/hazelcast/hazelcast-platform-operator/pull/371[#371])

Reviewer: ```suggestion
* Added support for the Topic CR (https://github.com/hazelcast/hazelcast-platform-operator/pull/371[#371])
```

> * Add support for ReplicatedMap CR (https://github.com/hazelcast/hazelcast-platform-operator/pull/394[#394])

Reviewer: ```suggestion
* Added support for the ReplicatedMap CR (https://github.com/hazelcast/hazelcast-platform-operator/pull/394[#394])
```

> * Add support for CronHotBackup CR (https://github.com/hazelcast/hazelcast-platform-operator/pull/362[#362])

Reviewer: ```suggestion
* Added support for the CronHotBackup CR (https://github.com/hazelcast/hazelcast-platform-operator/pull/362[#362])
```

> * Add support for restoring from HotBackup resource name (https://github.com/hazelcast/hazelcast-platform-operator/pull/401[#401])

Reviewer: ```suggestion
* Added support for restores using the HotBackup resource name (https://github.com/hazelcast/hazelcast-platform-operator/pull/401[#401])
```

> * Make Jet engine configurable from Hazelcast CR (https://github.com/hazelcast/hazelcast-platform-operator/pull/457[#457])

Reviewer: ```suggestion
* Added configuration for the Jet engine to the Hazelcast CR (https://github.com/hazelcast/hazelcast-platform-operator/pull/457[#457])
```

> * Add Map Entry listener to get notified when a map entry is added, removed, updated, evicted, or expired (https://github.com/hazelcast/hazelcast-platform-operator/pull/408[#408])

Reviewer: ```suggestion
* When a map entry is added, removed, updated, evicted, or expired, the Map Entry listener is notified (https://github.com/hazelcast/hazelcast-platform-operator/pull/408[#408])
```

Author: @angelasimms I'm not sure about it. 
We intentionally want to mention that it is a new feature. With the suggested change, it looks like we just changed/updated functionality. But before, we don't have a Map Entry listener.

Reviewer: @cheels would this be better. 

Added a Map Entry listener that is notified when a map entry is added, removed, updated, evicted, or expired (https://github.com/hazelcast/hazelcast-platform-operator/pull/408[#408])

Author: @angelasimms Yes, much better! Thanks!

> * Add support WAN Replication for multiple resources (https://github.com/hazelcast/hazelcast-platform-operator/pull/363[#363])

Reviewer: ```suggestion
* Add support for using WAN Replication with multiple resources (https://github.com/hazelcast/hazelcast-platform-operator/pull/363[#363])
```

> * Add ClearWanQueue function into WanReplication finalizer executor (https://github.com/hazelcast/hazelcast-platform-operator/pull/403[#403])

Reviewer: ```suggestion
* Added the `ClearWanQueue` function into the `WanReplication` finalizer executor to clean the WAN Replication queue after a CR deletion (https://github.com/hazelcast/hazelcast-platform-operator/pull/403[#403])
```

> * Make LOGGING_LEVEL configurable for Hazelcast containers (https://github.com/hazelcast/hazelcast-platform-operator/pull/352[#352])

Reviewer: ```suggestion
* Made the `LOGGING_LEVEL` configurable for Hazelcast containers (https://github.com/hazelcast/hazelcast-platform-operator/pull/352[#352])
```

== Changes

> * Remove the schedule from HotBackup API (https://github.com/hazelcast/hazelcast-platform-operator/pull/362[#362])

Reviewer: ```suggestion
* Removed scheduled backups from HotBackup API (https://github.com/hazelcast/hazelcast-platform-operator/pull/362[#362])
```

Reviewer: Moved scheduling of HotBackups from the HotBackup API to the CronHotBackup CR.

WDYT? @cheels

Author: It's ok. 👍 Thanks!

> * Add a default value to resource Kind in WanReplication spec (https://github.com/hazelcast/hazelcast-platform-operator/pull/374[#374])

Reviewer: ```suggestion
* Added a default value `kind` property value to the WanReplication CR specification (https://github.com/hazelcast/hazelcast-platform-operator/pull/374[#374])
```

Reviewer: Probably this is better: 

Added a default `kind` property value to the WanReplication CR specification

Author: @angelasimms Is it ok to have the next: 
`Added a default value to the `kind` property in the `WanReplication` CR specification.`

> * Add namespaced Role and RoleBinding that are created and passed to the namespaced service account of the hazelcast cluster to grant secret access (https://github.com/hazelcast/hazelcast-platform-operator/pull/407[#407])

Reviewer: ```suggestion
* Added a namespaced Role and RoleBinding that are created and passed to the namespaced service account of the Hazelcast cluster to grant secret access (https://github.com/hazelcast/hazelcast-platform-operator/pull/407[#407])
```

> * Replace REST client with Go for backup (https://github.com/hazelcast/hazelcast-platform-operator/pull/381[#381])

Reviewer: ```suggestion
* Replaced the  REST client with Go for triggering backups (https://github.com/hazelcast/hazelcast-platform-operator/pull/381[#381])
```

> * Change default repo to EE when licenseKeySecret is present (https://github.com/hazelcast/hazelcast-platform-operator/pull/402[#402])

Reviewer: ```suggestion
* Changed the default repository to Hazelcast Enterprise when the `licenseKeySecret` is present (https://github.com/hazelcast/hazelcast-platform-operator/pull/402[#402])
```

> * Use TLS for backup server sidecar (https://github.com/hazelcast/hazelcast-platform-operator/pull/350[#350])

Reviewer: ```suggestion
* Enables TLS for the backup server sidecar (https://github.com/hazelcast/hazelcast-platform-operator/pull/350[#350])
```

> * Add retry for backup service calls (https://github.com/hazelcast/hazelcast-platform-operator/pull/434[#434])

Reviewer: ```suggestion
* Added a retry for backup service calls (https://github.com/hazelcast/hazelcast-platform-operator/pull/434[#434])
```

> * Add WanReplication API Ref (https://github.com/hazelcast/hazelcast-platform-operator/pull/355[#355])

Reviewer: ```suggestion
* Added `WanReplication` to the API reference (https://github.com/hazelcast/hazelcast-platform-operator/pull/355[#355])
```

> * Enables TLS for the backup server sidecar (https://github.com/hazelcast/hazelcast-platform-operator/pull/350[#350])

Reviewer: ```suggestion
* Enabled TLS for the backup server sidecar (https://github.com/hazelcast/hazelcast-platform-operator/pull/350[#350])
```

> * Refactor Multimap CR and Topic CR data structure and Hazelcast configMap creation (https://github.com/hazelcast/hazelcast-platform-operator/pull/393[#393])

Reviewer: ```suggestion
* Refactored the Multimap and Topic CR data structures and Hazelcast's `configMap` creation (https://github.com/hazelcast/hazelcast-platform-operator/pull/393[#393])
```

> * Upgraded Hazelcast and MC versions to 5.1.4

Reviewer: ```suggestion
* Upgraded Hazelcast and Management Center versions to 5.1.4
```

== Fixed Issues

> * Fix topic controller import (https://github.com/hazelcast/hazelcast-platform-operator/pull/387[#387])

Reviewer: ```suggestion
* Fixes topic controller import (https://github.com/hazelcast/hazelcast-platform-operator/pull/387[#387])
```

> * Fix error hiding in startBackup() function (https://github.com/hazelcast/hazelcast-platform-operator/pull/445[#445])

Reviewer: ```suggestion
* Errors are now handled correctly by the `startBackup()` function. Previously, no errors were returned (https://github.com/hazelcast/hazelcast-platform-operator/pull/445[#445])
```

> * Remove the agent port from the discovery service, which fixes the problem when the member can't get an external IP (https://github.com/hazelcast/hazelcast-platform-operator/pull/409[#409])

Reviewer: ```suggestion
* Removed the agent port from the Kubernetes discovery service, to resolve an issue where external IP addresses were not correctly allocated to members (https://github.com/hazelcast/hazelcast-platform-operator/pull/409[#409])
```
